lib/schemas.py

```diff
@@ -32,34 +36,50 @@ class GenericItem(BaseModel):
     text: Optional[str] = None
     raw: Optional[Dict] = {}
     parameters: Optional[Dict] = {}
-    result: Optional[Union[ErrorResponse, MediaResponse, VideoResponse, YakeKeywordsResponse, ClassyCatSchemaResponse, ClassyCatBatchClassificationResponse]] = None
+    result: Optional[Any] = None
 
 class Message(BaseModel):
     body: GenericItem
     model_name: str
     retry_count: int = 0
 
-def parse_message(message_data: Dict) -> Message:
+def parse_input_message(message_data: Dict) -> Message:
     body_data = message_data['body']
     model_name = message_data['model_name']
     result_data = body_data.get('result', {})
-    if 'yake_keywords' in model_name:
-        result_instance = YakeKeywordsResponse(**result_data)
-    elif 'classycat' in model_name:
-        event_type = body_data['parameters']['event_type']
-        if event_type == 'classify':
-            result_instance = ClassyCatBatchClassificationResponse(**result_data)
-        elif event_type == 'schema_lookup' or event_type == 'schema_create':
-            result_instance = ClassyCatSchemaResponse(**result_data)
+
+    modelClass = get_class('lib.model.', os.environ.get('MODEL_NAME'))
```

Comment:
why does this need to be different than the `model_name` associated with the message?  do the names not quite align?

Reply:
that's a great question. I copied this from `create()` in `lib.model.model.py`

but I agree that it makes more sense to use the model_name mentioned in the input and will update the code.

this function has also some other not so great logic that needs to be refactored down the line, such as assuming every model_name input that is not video or yake will return a MediaResponse. I have outlined my ideas for this work in a separate ticket https://meedan.atlassian.net/browse/CV2-5093